astroplan/constraints.py

```diff
@@ -0,0 +1,614 @@
+
+        Parameters
+        ----------
+        times : `~astropy.time.Time`
+            Array of times on which to test the constraint
+
+        targets : {list, `~astropy.coordinates.SkyCoord`, `~astroplan.FixedTarget`}
+            Target or list of targets
+
+        observer : `~astroplan.Observer`
+            The observer who has constraints ``constraints``
+
+        time_resolution : `~astropy.units.Quantity` (optional)
+            Set the time resolution in calculations of the altitude/azimuth
+
+        Returns
+        -------
+        altaz_dict : dict
+            Dictionary containing two key-value pairs. (1) 'times' contains the
+            times for the alt/az computations, (2) 'altaz' contains the
+            corresponding alt/az coordinates at those times.
+        """
+        if not hasattr(observer, '_altaz_cache'):
+            observer._altaz_cache = {}
+
+        # convert times, targets to tuple for hashing
+        aakey = (tuple(times.jd), tuple(targets))
+
+        if aakey not in observer._altaz_cache:
+            try:
+                if force_zero_pressure:
+                    observer_old_pressure = observer.pressure
+                    observer.pressure = 0
+
+                altaz = observer.altaz(times, targets)
+                observer._altaz_cache[aakey] = dict(times=times,
+                                                    altaz=altaz)
+            finally:
+                if force_zero_pressure:
+                    observer.pressure = observer_old_pressure
+
+        return observer._altaz_cache[aakey]
+
+class AltitudeConstraint(Constraint):
+    """
+    Constrain the altitude of the target.
+
+    .. note::
+        This will misbehave if you try to constrain negative altitudes, as
+        the `~astropy.coordinates.AltAz` frame tends to mishandle negative
+        altitudes.
+    """
+    def __init__(self, min=None, max=None):
+        """
+        Parameters
+        ----------
+        min : `~astropy.units.Quantity` or `None`
+            Minimum altitude of the target. `None` indicates no limit.
+
+        max : `~astropy.units.Quantity` or `None`
+            Maximum altitude of the target. `None` indicates no limit.
+        """
+        if min is None:
+            self.min = 0*u.deg
+        else:
+            self.min = min
+        if max is None:
+            self.max = 90*u.deg
+        else:
+            self.max = max
+
+    def compute_constraint(self, times, observer, targets):
+
+        cached_altaz = self._get_altaz(times, observer, targets)
+        altaz = cached_altaz['altaz']
+        lowermask = self.min < altaz.alt
+        uppermask = altaz.alt < self.max
+        return lowermask & uppermask
+
+class AirmassConstraint(AltitudeConstraint):
+    """
+    Constrain the airmass of a target.
+
+    In the current implementation the airmass is approximated by the secant of
+    the zenith angle.
+    """
+    def __init__(self, max=None, min=None):
```

Comment:
I agree it is ok to ignore the IDE error: it's true that you're not calling the superclass `__init__`, but that doesn't matter here because the superclass `__init__`s aren't supposed to be doing anything anyway.  It *could* end up confusing if any user wanted to subclass `AirmassConstraint`, but I can't imagine why someone would actually want to do that...

In fact, as it's *currently* implemented, it's not really necessary for `AirmassConstraint ` to be an `AltitudeConstraint` subclass, because you use *none* of the`AltitudeConstraint` behavior in  `AirmassConstraint `.  The main reason to keep it this way is conceptual: I think people think of airmass as sort of another way of thinking about altitude/zentih angle. It also might make more sense down the road for it to be a subclass if more agressive caching of something or other happens: you can implement it all in `AltitudeConstraint` and have `AirmassConstraint` just do a thin layer that converts an airmass into an altitude.  But for now leaving it as-is seems just fine.